datafusion/physical-expr/src/planner.rs

```diff
@@ -310,6 +313,8 @@ pub fn create_physical_expr(
                                 &list_expr_data_type,
                                 &value_expr_data_type,
                             ) {
+                                // TODO: Can't cast from list type to value type directly
```

Comment:
we can't cast the list data type to the value data type directly, and need to coerced common data type if they are compatible.

Reply:
about issue: https://github.com/apache/arrow-datafusion/issues/2759